evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -12,22 +12,28 @@
 
 
 def test_stl_decomposer_init():
-    delayed_features = STLDecomposer(degree=3, time_index="dates")
-    assert delayed_features.parameters == {
+    decomp = STLDecomposer(degree=3, time_index="dates")
+    assert decomp.parameters == {
         "degree": 3,
-        "seasonal_period": 7,
+        "period": None,
+        "seasonal_smoother": 7,
         "time_index": "dates",
     }
 
 
-def test_stl_decomposer_auto_sets_seasonal_period_to_odd(ts_data):
-    X, _, y = ts_data()
+def test_stl_decomposer_auto_sets_seasonal_smoother_to_odd():
+    stl = STLDecomposer(seasonal_smoother=3)
+    assert stl.seasonal_smoother == 3
+
+    stl = STLDecomposer(seasonal_smoother=4)
+    assert stl.seasonal_smoother == 5
 
-    stl = STLDecomposer(seasonal_period=3)
-    assert stl.seasonal_period == 3
 
-    stl = STLDecomposer(seasonal_period=4)
-    assert stl.seasonal_period == 5
+def test_stl_raises_warning_high_smoother(caplog, ts_data):
+    X, _, y = ts_data()
+    stl = STLDecomposer(seasonal_smoother=101)
+    stl.fit(X, y)
+    assert "STLDecomposer may perform poorly" in caplog.text
```

Comment:
I suspect the behavior we want here is actually to warn off the `period` rather than the `seasonal_smoother` parameter here.